src/librustdoc/html/render/mod.rs

```diff
@@ -4419,6 +4382,74 @@ fn sidebar_assoc_items(it: &clean::Item) -> String {
     out
 }
 
+fn sidebar_deref_methods(impl_: &Impl, v: &Vec<Impl>) -> String {
+    let mut out = String::new();
+    let c = cache();
+
+    if let Some((target, real_target)) =
+        impl_.inner_impl().items.iter().find_map(|item| match *item.kind {
+            clean::TypedefItem(ref t, true) => Some(match *t {
+                clean::Typedef { item_type: Some(ref type_), .. } => (type_, &t.type_),
+                _ => (&t.type_, &t.type_),
+            }),
+            _ => None,
+        })
+    {
+        let deref_mut = v
+            .iter()
+            .filter(|i| i.inner_impl().trait_.is_some())
+            .any(|i| i.inner_impl().trait_.def_id() == c.deref_mut_trait_did);
+        let inner_impl = target
+            .def_id()
+            .or_else(|| {
+                target.primitive_type().and_then(|prim| c.primitive_locations.get(&prim).cloned())
+            })
+            .and_then(|did| c.impls.get(&did));
+        if let Some(impls) = inner_impl {
+            let id = small_url_encode(&format!("deref-methods-target-{:#}", real_target.print()));
+            out.push_str(&format!("<a class=\"sidebar-title\" href=\"#{}\">", id));
```

Comment:
We tend to generally make one big `push_str` call over a little ones. In my opinion, it makes code reading simpler but maybe you (and others) have different opinions on this subject?

Reply:
I think it's easier to read this way, since generating the id does non-trivial work.

Reply:
You can still have the id generated separately. I was pointing out all the `push_str` calls here. ;)

Reply:
Ah, I have no strong opinion about the multiple `push_str`s here... actually, this is just existing code I extracted out to separate function. 😅 Anyway, I'm happy to fold the `push_str`s together.